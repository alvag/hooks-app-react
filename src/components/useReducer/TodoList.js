import React from 'react';
import TodoListItem from './TodoListItem';

const TodoList = ( { todos, deleteTodo, toggleTodo } ) => {
    return (
        <ul className="list-group list-group-flush">
            {
                todos.map( ( todo, index ) => (
                    <TodoListItem key={todo.id}
                                  todo={todo}
                                  deleteTodo={deleteTodo}
                                  toggleTodo={toggleTodo}
                                  index={index + 1}/>
                ) )
            }
        </ul>
    );
};

export default TodoList;