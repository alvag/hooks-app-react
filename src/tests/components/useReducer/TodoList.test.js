import React from 'react';
import { shallow } from 'enzyme';
import TodoList from '../../../components/useReducer/TodoList';
import { demoTodos } from '../../fixtures/demoTodos';

describe( 'Pruebas en <TodoList />', function() {

    const deleteTodo = jest.fn();
    const toggleTodo = jest.fn();

    const wrapper = shallow( <
            TodoList
            todos={demoTodos}
            deleteTodo={deleteTodo}
            toggleTodo={toggleTodo}
        />
    );

    test( 'debe mostrarse correctamente', () => {
        expect( wrapper ).toMatchSnapshot();
    } );

    test( 'debe tener dos <TodoListItem />', () => {
        expect( wrapper.find( 'TodoListItem' ).length ).toBe( demoTodos.length );

        expect( wrapper.find( 'TodoListItem' ).at( 0 ).prop( 'deleteTodo' ) ).toEqual( expect.any( Function ) );
    } );

} );