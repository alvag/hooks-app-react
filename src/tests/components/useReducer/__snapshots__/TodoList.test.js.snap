// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <TodoList /> debe mostrarse correctamente 1`] = `
<ul
  className="list-group list-group-flush"
>
  <TodoListItem
    deleteTodo={[MockFunction]}
    index={1}
    key="1"
    todo={
      Object {
        "description": "Aprender React",
        "done": false,
        "id": 1,
      }
    }
    toggleTodo={[MockFunction]}
  />
  <TodoListItem
    deleteTodo={[MockFunction]}
    index={2}
    key="2"
    todo={
      Object {
        "description": "Aprender Javascript",
        "done": false,
        "id": 2,
      }
    }
    toggleTodo={[MockFunction]}
  />
</ul>
`;
